iotilesensorgraph/iotile/sg/graph.py

```diff
@@ -514,3 +559,102 @@ def dump_streamers(self):
+            constant_dump.append("'{}' {}".format(stream, value))
+
+        return constant_dump
+
+    def dump_metadata_database(self):
+        """Dump all of the metadata in this sensor graph as a list of strings."""
+
+        metadata_dump = []
+
+        for metadata, value in sorted(self.metadata_database.items(), key=lambda x: x[0].encode()):
+            metadata_dump.append("{}: {}".format(metadata, value))
+
+        return metadata_dump
+
+    def dump_config_database(self, dump_config_type=True, ignore_configs=[]):
+        """Dump all of the config variables in this sensor graph as a list of strings."""
+
+        config_dump = []
+        for slot, conf_vars in self.config_database.items():
+            for conf_var, conf_def in conf_vars.items():
+                conf_type, conf_val = conf_def
+
+                if conf_var in ignore_configs:
+                    continue
+
+                if dump_config_type:
+                    config_dump.append("'{}' {} {} {}".format(slot, conf_var,
+                                                              conf_type,
+                                                              conf_val))
+                else:
+                    config_dump.append("'{}' {} {}".format(slot, conf_var,
+                                                              conf_val))
+
+        return config_dump
+
+    def get_nodes_binary(self):
+        """Returns the binary representation of all the nodes"""
+
+        from .node_descriptor import create_binary_descriptor
+        binary_representation = bytearray()
+
+        for node in self.dump_nodes():
+            binary_representation += create_binary_descriptor(node)
+
+        return binary_representation
+
+    def get_streamers_binary(self):
+        """Returns the binary representation of all the streamers"""
+
+        from .streamer_descriptor import parse_string_descriptor, create_binary_descriptor
+        binary_representation = bytearray()
+
+        for streamer in self.dump_streamers():
+            streamer_obj = parse_string_descriptor(streamer)
+            binary_representation += create_binary_descriptor(streamer_obj)
+
+        return binary_representation
+
+    def get_constant_database_binary(self):
+        """Returns the binary representation of all the constant streams"""
+
+        binary_representation = bytearray()
+
+        for constant in self.dump_constant_database():
+            binary_representation += bytes(constant, 'utf-8')
+
+        return binary_representation
+
+    def get_metadata_database_binary(self):
+        """Returns the binary representation of the KNOWN metadata variables"""
+
+        binary_representation = bytearray()
+
+        for metadata in self.dump_metadata_database():
+            entry, _ = metadata.split(': ')
+
+            if entry in known_metadata:
+                binary_representation += bytes(metadata, 'utf-8')
```

Comment:
At this moment, we only want to support the two known meta tags, `app_tag` and `app_version`.